test-suite/blackformula.cpp

```diff
@@ -276,6 +277,182 @@ void BlackFormulaTest::testImpliedVolAdaptiveSuccessiveOverRelaxation() {
     }
 }
 
+void assertBlackFormulaForwardDerivative(
+    Option::Type optionType,
+    const std::vector<Real> &strikes,
+    Real bpvol)
+{
+    Real forward = 1.0;
+    Real tte = 10.0;
+    Real stdDev = bpvol * std::sqrt(tte);
+    Real discount = 0.95;
+    Real displacement = 0.01;
+    Real bump = 0.0001;
+    Real epsilon = 1.e-10;
+
+    for (std::vector<Real>::const_iterator it = strikes.begin(); it != strikes.end(); ++it)
+    {
+        Real strike = *it;
+        Real delta = blackFormulaForwardDerivative(
+            optionType, strike, forward, stdDev, discount, displacement);
+        Real bumpedDelta = blackFormulaForwardDerivative(
+            optionType, strike, forward + bump, stdDev, discount, displacement);
+
+        Real basePremium = blackFormula(
+            optionType, strike, forward, stdDev, discount, displacement);
+        Real bumpedPremium = blackFormula(
+            optionType, strike, forward + bump, stdDev, discount, displacement);
+        Real deltaApprox = bumpedPremium - basePremium;
+
+        /*! Based on the Mean Value Theorem, the below inequality
+            should hold for any function that is monotonic in the
+            area of the bump.
+         */
+        bool success = (std::max(delta * bump, bumpedDelta * bump) + epsilon > (
+            deltaApprox) > std::min(delta * bump, bumpedDelta * bump) - epsilon);
```

Comment:
`a > b > c` doesn't work in C++ as you would expect&mdash;instead, the boolean `(a>b)` is converted to `int` and compared to `c`.
```suggestion
        bool success =
            (std::max(delta * bump, bumpedDelta * bump) + epsilon > deltaApprox)
            &&  (deltaApprox > std::min(delta * bump, bumpedDelta * bump) - epsilon);
```